helix-core/src/test/java/org/apache/helix/TestHelper.java

```diff
@@ -847,4 +850,161 @@ public static void printZkListeners(HelixZkClient client) throws Exception {
     }
     System.out.println("}");
   }
+
+  // Thread dump related utility function
+  private static ThreadGroup getRootThreadGroup() {
+    ThreadGroup candidate = Thread.currentThread().getThreadGroup();
+    while (candidate.getParent() != null) {
+      candidate = candidate.getParent();
+    }
+    return candidate;
+  }
+
+  public static List<Thread> getAllThreads() {
+    ThreadGroup rootThreadGroup = getRootThreadGroup();
+    Thread[] threads = new Thread[32];
+    int count = rootThreadGroup.enumerate(threads);
+    while (count == threads.length) {
+      threads = new Thread[threads.length * 2];
+      count = rootThreadGroup.enumerate(threads);
+    }
+    return Arrays.asList(Arrays.copyOf(threads, count));
+  }
+
+  static public enum ThreadCategory {
+    ZkServer("zookeeper server threads", 4, 100,
```

Comment:
The hard coded patterns seem hard to maintain. If any thread names are changed, then their leakage won't be detected, right? Can we make the pattern string defined in some constants and refer to them in here and also in the place where thread names are specified?

Reply:
done.